util/settings/settings.go

```diff
@@ -2317,6 +2320,26 @@ func (mgr *SettingsManager) GetExcludeEventLabelKeys() []string {
 	return labelKeys
 }
 
+func (mgr *SettingsManager) GetHideSecretAnnotations() map[string]bool {
+	annotsKeys := make(map[string]bool)
+	argoCDCM, err := mgr.getConfigMap()
+	if err != nil {
+		log.Error(fmt.Errorf("failed getting configmap: %v", err))
+		return nil
+	}
+	if value, ok := argoCDCM.Data[hideSecretAnnotations]; ok {
+		annots := strings.Split(value, "\n")
+		for _, a := range annots {
+			a = regexp.MustCompile(`^\s*-`).ReplaceAllString(a, "")
```

Comment:
Is there a kubernetes library function we can use to make sure the annotations follow the Kubernetes guidelines instead?

Reply:
The regex was used to clean up the input annotations from configmap and not validate them. I have removed the regex by updated the code to take in comma-seperated list of annotation keys from configmap instead of a list.